src/fit.jl

```diff
@@ -21,10 +23,13 @@ array that holds the validation metric's value at each iteration.
 * `weights::Vector{Tw<:Real}`: the training weights.
 * `init_score::Vector{Ti<:Real}`: the init scores.
 """
-function fit!(estimator::LGBMEstimator, X::Matrix{TX},
-    y::Vector{Ty}, test::Tuple{Matrix{TX},Vector{Ty}}...; verbosity::Integer = 1,
-    is_row_major = false, weights::Vector{Tw} = Vector{Float32}(),
-    init_score::Vector{Ti} = Vector{Float64}()) where {TX<:Real,Ty<:Real,Tw<:Real,Ti<:Real}
+function fit!(
+    estimator::LGBMEstimator, num_iterations::Int, X::Matrix{TX}, y::Vector{Ty}, test::Tuple{Matrix{TX},Vector{Ty}}...;
```

Comment:
Needs some thought, I am not sure chucking in num_iterations here like this is sensible

If we're updating an existing booster, we don't want to go through the rigmarole of setting up the dataset again